### PR TITLE
Remove fixed_auto_queue_size threadpool type

### PR DESCRIPTION
**Pull request by ywelsch**
The `fixed_auto_queue_size` thread pool holds a fixed size of threads to handle the requests with a bounded queue for pending requests that have no threads to service them. It's similar to the `fixed` threadpool, however, the `queue_size` automatically adjusts according to calculations based on
[Little's Law](https://en.wikipedia.org/wiki/Little%27s_law). This dynamic adjustment was, by default, disabled for the two `fixed_auto_queue_size` thread pools (`search` and `search_throttled`).

As this feature was marked as experimental, and we are unaware of any users explicitly configuring this, we intent to deprecate it in ES 7 and remove it in ES 8.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/query/QueryPhase.java [hunk 1]
```diff
@@ -300,12 +297,6 @@ static boolean executeInternal(SearchContext searchContext) throws QueryPhaseExe
                 restoreTopFieldDocs(queryResult, sortAndFormatsForRewrittenNumericSort);
             }
 
-            ExecutorService executor = searchContext.indexShard().getThreadPool().executor(ThreadPool.Names.SEARCH);
-            if (executor instanceof QueueResizingEsThreadPoolExecutor) {
-                QueueResizingEsThreadPoolExecutor rExecutor = (QueueResizingEsThreadPoolExecutor) executor;
-                queryResult.nodeQueueSize(rExecutor.getCurrentQueueSize());
-                queryResult.serviceTimeEWMA((long) rExecutor.getTaskExecutionEWMA());
-            }
```
Comment (dakrone) [inline server/src/main/java/org/elasticsearch/search/query/QueryPhase.java]: These stats are still used for adaptive replica selection. This is what I meant when we discussed this about the `QueueResizingEsThreadPoolExecutor` working for both the little's law part (the part we wanted to remove) as well as for adaptive replica selection.

**Review by dakrone — CHANGES_REQUESTED** [review]
This removes too much, we still need a lot of the infrastructure from `QueueResizingEsThreadPoolExecutor` to gather the metrics required for adaptive replica selection.

File: server/src/main/java/org/elasticsearch/search/query/QueryPhase.java (hunk 1 above)
Comment (ywelsch) [reply server/src/main/java/org/elasticsearch/search/query/QueryPhase.java]: You're absolutely right. I was too eager to delete code here. Will be interesting to see if any test catches this. Will rework this PR tomorrow.

Comment (ywelsch) [reply server/src/main/java/org/elasticsearch/search/query/QueryPhase.java]: I've addressed this now in 382d40b

**Comment by ywelsch** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-matrix-windows

**Review by dakrone — APPROVED** [review]
LGTM, thanks Yannick

**Pull request merged**